Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -50,6 +50,10 @@ extension SQLiteHistory: HistoryRecommendations {
             "       ORDER BY visitDate DESC" +
             "   )" +
             "   LEFT JOIN \(TableHistory) ON \(TableHistory).id = s" +
+            "   INNER JOIN (" +
+            "       SELECT \(TableHistory).domain_id AS domain_id, MAX(\(TableHistory).id) AS history_id FROM \(TableHistory)" +
+            "       GROUP BY \(TableHistory).domain_id" +
+            "   ) AS domains ON domains.domain_id = \(TableHistory).domain_id AND \(TableHistory).id = history_id" +
```

Comment:
Using `MAX` here for the query to return a unique, predictable result makes sense but I think we might be able to improve this by using the last visit date instead since ID is somewhat arbitrary. Turns out we already have a view named `ViewHistoryVisits` that we can query that will tell us this. The only thing missing from the view is the domain ID but I believe we can surface that in the view from the history table we're joining with. When adding the column to the view we'll need to make sure in the migration that we DROP the old view and recreate it. We do this a few times throughout our versions so you can see how we do it there.

On a minor note, since we're using the same derived table subquery in both the `nonRecentHistory` and `bookmarkHighlights`, we can DRY this up by moving the query string into a variable and inserting it in the query.